src/core/CMakeLists.txt

```diff
@@ -56,15 +56,8 @@ set( SOURCES_CPP
      utilities/mpi_utils.cpp
     )
 
-find_package(MPI)
-
-add_library(${PROJECT_NAME}  ${SOURCES_INC} ${SOURCES_CPP})
-target_compile_options(${PROJECT_NAME}  PRIVATE ${PHARE_WERROR_FLAGS})
-target_link_libraries(${PROJECT_NAME}  PRIVATE phare_initializer ${MPI_C_LIBRARIES} ${MPI_CXX_LIBRARIES}
```

Comment:
I have reproduced the mpicxx link error locally and have verified that this removal of `${MPI_CXX_LIBRARIES}` is in fact what is causing the error.

We were always including the mpicxx library before in this lib, or that use this lib, which we don't need anymore